microk8s-resources/actions/common/utils.sh

```diff
@@ -397,19 +397,40 @@ get_ips() {
     fi
 }
 
-gen_server_cert() (
+_gen_server_cert() {
     export OPENSSL_CONF="/snap/microk8s/current/etc/ssl/openssl.cnf"
     ${SNAP}/usr/bin/openssl req -new -sha256 -key ${SNAP_DATA}/certs/server.key -out ${SNAP_DATA}/certs/server.csr -config ${SNAP_DATA}/certs/csr.conf
     ${SNAP}/usr/bin/openssl x509 -req -sha256 -in ${SNAP_DATA}/certs/server.csr -CA ${SNAP_DATA}/certs/ca.crt -CAkey ${SNAP_DATA}/certs/ca.key -CAcreateserial -out ${SNAP_DATA}/certs/server.crt -days 365 -extensions v3_ext -extfile ${SNAP_DATA}/certs/csr.conf
-)
+}
 
-gen_proxy_client_cert() (
+gen_server_cert() {
+    # Generate the server certificate
+    # The file csr.conf is checked to determine if a regeneration of the cert must be done.
+    if compare_csr_conf; then
+        _gen_server_cert
+    fi
+}
+
+_gen_proxy_client_cert() {
```

Comment:
I am not sure we are doing the right thing here. 

Let's say someone wants to reproduce the server certificates. The refresh_certs.py script in [1] will `rm server.crt` and then will call this `gen_server_cert` function. If there is no change in the `csr.conf.template` (which is expected) we will not be recreating the server.crt. 


[1] https://github.com/canonical/microk8s/blob/master/scripts/wrappers/refresh_certs.py#L139

Reply:
Yes correct. I would propose to remove the csr.conf file in all the relevant functions in `refresh_certs.py` as it is done for the function _reproduce_all_root_ca_certs_: https://github.com/canonical/microk8s/blob/master/scripts/wrappers/refresh_certs.py#L110=
WDYT?